src/helpers/getLedger.ts

```diff
@@ -1,19 +1,24 @@
-import { QUERY_BLOCK_LIMIT } from '@big-whale-labs/constants'
 import { SealCredLedger } from '@big-whale-labs/seal-cred-ledger-contract'
 import Ledger from 'models/Ledger'
 import getLedgerRecord from 'helpers/getLedgerRecord'
+import getAllEvents from './getAllEvents'
 
 export default async function (sealCredLedger: SealCredLedger) {
-  const eventsFilter = sealCredLedger.filters.SetMerkleRoot()
-  const events = await sealCredLedger.queryFilter(
-    eventsFilter,
-    QUERY_BLOCK_LIMIT
-  )
+  const { events, deleteTopic } = await getAllEvents(sealCredLedger)
+
   const ledger = {} as Ledger
   const addressToMerkle: { [address: string]: string } = {}
 
   for (const event of events) {
-    const { tokenAddress, merkleRoot } = event.args
+    const {
+      args: { tokenAddress, merkleRoot },
+      topic,
+    } = event
+
+    if (topic === deleteTopic) {
```

Comment:
Maybe we can do: 
```TS
if (topic === deleteTopic) return delete addressToMerkle[tokenAddress]
```
Not sure if it will work with `delete`

Reply:
Return should work as `countinue` as I know

Reply:
@T-Damer `continue` ends the current loop, it doesn't add all loops